density.c

```diff
@@ -418,8 +418,9 @@ density_postprocess(int i, TreeWalk * tw)
                 SPHP(i).DhsmlEgyDensityFactor *= -SPHP(i).DhsmlDensityFactor;
                 SPHP(i).EgyWtDensity /= EntPred;
             } else {
-                SPHP(i).DhsmlEgyDensityFactor=0;
-                SPHP(i).EgyWtDensity=0;
+                /*Use non-weighted densities for this*/
+                SPHP(i).DhsmlEgyDensityFactor=SPHP(i).DhsmlDensityFactor;
+                SPHP(i).EgyWtDensity=SPHP(i).Density;
```

Comment:
Doesn't it usually signals a bigger problem? How often do you see it zero? Perhaps add a comment about this?

Reply:
I used it to find the bug in b5d1d95, but once that was fixed it is zero only on the first iteration of the first timestep. If we could make the first iteration non-zero it might indeed be better to do endrun() here instead, but I'm not sure how to do that. I'll add a comment.